### PR TITLE
platform-bukkit: fix ViaVersion softDepend occurring too late

### PR DESCRIPTION
**Pull request by TehBrian**

### CODE REVIEW

File: platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/BukkitAudiencesImpl.java [hunk 1]
```diff
@@ -134,6 +133,7 @@ static BukkitAudiences instanceFor(final @NotNull Plugin plugin) {
 
   @Override
   protected @NotNull BukkitAudience createAudience(final @NotNull Collection<CommandSender> viewers) {
+    this.softDepend("ViaVersion");
```
Comment (zml2008) [inline platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/BukkitAudiencesImpl.java:136]: This doesn't need to run on every audience creation -- maybe put it in the computeIfAbsent lambda within the build() method of the BukkitAudiences.Builder?

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Comment by kezz** [comment]
Would it not be easier to just lazily initialise the empty audience? It's the creation of the empty audience that is causing this issue after all.

**Comment by TehBrian** [comment]
Chucking softDepend in BukkitAudiences.Builder would fix the root problem (creating BukkitAudiencesImpl before calling softDepend) rather than the symptom, but both fixes would work. Let me know what fix to implement and I'll update the pr.

**Comment by kezz** [comment]
@TehBrian would you mind making *both* changes? So, moving the soft depend method *and* making the empty audience lazily initialised? Sorry for the annoyance :p

**Review by zml2008 — APPROVED** [review]

**Pull request merged**